### ISSUE TITLE
[UseOffsetPaging] does not mark the generated return type as 'not nullable'

### ISSUE THREAD

**Issue by RoystonS** [ChilliCream/graphql-platform#6430]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

When I mark a resolver method with `[UseOffsetPaging]`, I expect it to generate a query with a non-nullable return type. Instead, it isn't marked as nullable, forcing clients to have to deal with the possibility that they'll get null when requesting the paged data.

### Steps to reproduce

Here's a tiny repo that demonstrates the issue:

https://github.com/RoystonS/HotChocolatePagingSegmentNullability

The key piece of code is a resolver function that uses `[UseOffsetPaging]`:

```csharp
public class Root
{
    [UseOffsetPaging(IncludeTotalCount = true, MaxPageSize = 100, DefaultPageSize = 100)]
    public IQueryable<Value> GetValues([Parent] Root root)
    {
        return new List<Value>().AsQueryable();
    }
}
```

The generated schema looks like this:

```graphql
type Root {
  values(skip: Int, take: Int): ValuesCollectionSegment
}
```

I'd very much prefer to see:
```graphql
type Root {
  values(skip: Int, take: Int): ValuesCollectionSegment!
}
```

As it is, clients see a 'potentially null ValuesCollectionSegment' and have to deal with the possibility of it returning `null` when requested.

### Version

13.4.0

**Comment by michaelstaib** [comment]
Thats correct and intend to allow for error termination in the field. We will soon allow for people to override this.

**Comment by RoystonS** [comment]
@NicolyLuisaG, `ValueCollectionSegment` _isn't_ something I created: that type is created by HotChocolate itself. The code I've provided in this issue _is_ my entire relevant code. (There's a full example in the linked GitHub repo.)

I can't use the above suggested code because I don't have access to `ValueCollectionSegment` at compile time because whatever implementation there is is created dynamically at runtime by HotChocolate via its UseOffsetPaging attribute.

@michaelstaib To clarify, are you saying that the behaviour I'm observing is as currently intended, but an override mechanism is on the way?
